config/yaml_config.go

```diff
@@ -263,25 +95,149 @@ func mergeNetworkYamlConfig(agentConf *AgentConfig, networkConf *YamlAgentConfig
+
+	// Strips all process arguments
+	if config.Datadog.GetBool(key(ns, "strip_proc_arguments")) {
+		a.Scrubber.StripAllArguments = true
+	}
+
+	// How many check results to buffer in memory when POST fails. The default is usually fine.
+	if queueSize := config.Datadog.GetInt(key(ns, "queue_size")); queueSize > 0 {
+		a.QueueSize = queueSize
+	}
+
+	// The maximum number of file descriptors to open when collecting net connections.
+	// Only change if you are running out of file descriptors from the Agent.
+	if maxProcFDs := config.Datadog.GetInt(key(ns, "max_proc_fds")); maxProcFDs > 0 {
+		a.MaxProcFDs = maxProcFDs
+	}
+
+	// The maximum number of processes, or containers per message. Note: Only change if the defaults are causing issues.
+	if maxPerMessage := config.Datadog.GetInt(key(ns, "max_per_message")); maxPerMessage > 0 {
+		if maxPerMessage <= maxPerMessage {
+			a.MaxPerMessage = maxPerMessage
+		} else {
+			log.Warn("Overriding the configured item count per message limit because it exceeds maximum")
+		}
+	}
+
+	// Overrides the path to the Agent bin used for getting the hostname. The default is usually fine.
+	a.DDAgentBin = defaultDDAgentBin
+	if agentBin := config.Datadog.GetString(key(ns, "dd_agent_bin")); agentBin != "" {
+		a.DDAgentBin = agentBin
+	}
+
+	// Windows: Sets windows process table refresh rate (in number of check runs)
+	if argRefresh := config.Datadog.GetInt(key(ns, "windows", "args_refresh_interval")); argRefresh != 0 {
+		a.Windows.ArgsRefreshInterval = argRefresh
+	}
+
+	// Windows: Controls getting process arguments immediately when a new process is discovered
+	if addArgsKey := key(ns, "windows", "add_new_args"); config.Datadog.IsSet(addArgsKey) {
+		a.Windows.AddNewArgs = config.Datadog.GetBool(addArgsKey)
 	}
 
-	// load the configuration
-	if err := ddconfig.Load(); err != nil {
-		return fmt.Errorf("unable to load Datadog config file: %s", err)
+	// Optional additional pairs of endpoint_url => []apiKeys to submit to other locations.
+	for endpointURL, apiKeys := range config.Datadog.GetStringMapStringSlice(key(ns, "additional_endpoints")) {
+		u, err := URL.Parse(endpointURL)
+		if err != nil {
+			return fmt.Errorf("invalid additional endpoint url '%s': %s", endpointURL, err)
+		}
+		for _, k := range apiKeys {
+			a.APIEndpoints = append(a.APIEndpoints, APIEndpoint{
+				APIKey:   k,
+				Endpoint: u,
+			})
+		}
 	}
 
+	// Pull additional parameters from the global config file.
+	a.LogLevel = config.Datadog.GetString("log_level")
+	a.StatsdPort = config.Datadog.GetInt("dogstatsd_port")
+	a.Transport = ddutil.CreateHTTPTransport()
+
 	return nil
 }
+
+func (a *AgentConfig) setCheckInterval(ns, check, checkKey string) {
+	if interval := config.Datadog.GetInt(key(ns, "intervals", check)); interval != 0 {
+		log.Infof("Overriding container check interval to %ds", interval)
+		a.CheckIntervals[checkKey] = time.Duration(interval) * time.Second
+	}
```

Comment:
should we log something if the interval is 0 ?

Reply:
No - default would be zero so we'd always see that log message